src/python/tests/genome_metadata/test_prepare.py

```diff
@@ -0,0 +1,344 @@
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+from pathlib import Path
+from typing import Any, Callable, ContextManager, Dict, Optional
+from unittest.mock import Mock, patch
+from xml.etree import ElementTree
+
+from deepdiff import DeepDiff
+import pytest
+
+from ensembl.io.genomio.genome_metadata import prepare
+
+
+# @pytest.mark.dependency(name="test_get_gbff_regions")
+@pytest.mark.parametrize(
+    "genome_file, gff3_file, output, expectation",
+    [
+        pytest.param(
+            "genbank_genome.json",
+            None,
+            {
+                "assembly": {
+                    "provider_name": "GenBank",
+                    "provider_url": "https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_013436015.2",
+                },
+            },
+            does_not_raise(),
+            id="GenBank assembly",
+        ),
+        pytest.param(
+            "refseq_genome.json",
+            Path("fake.gff3"),
+            {
+                "assembly": {
+                    "provider_name": "RefSeq",
+                    "provider_url": "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000004695.1",
+                },
+                "annotation": {
+                    "provider_name": "RefSeq",
+                    "provider_url": "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000004695.1",
+                },
+            },
+            does_not_raise(),
+            id="RefSeq assembly and annotation",
+        ),
+        pytest.param(
+            "updated_genome.json",
+            Path("fake.gff3"),
+            {
+                "assembly": {"provider_name": "GenBank", "provider_url": None},
+                "annotation": {"provider_name": "GenBank", "provider_url": None},
+            },
+            does_not_raise(),
+            id="Provider information already present",
+        ),
+        pytest.param(
+            "cncb_genome.json", None, {}, pytest.raises(prepare.MetadataError), id="Unexpected provider"
+        ),
+    ],
+)
+def test_add_provider(
+    json_data: Callable[[str], Any],
+    genome_file: str,
+    gff3_file: Optional[Path],
+    output: Dict[str, Dict[str, Optional[str]]],
+    expectation: ContextManager,
+) -> None:
+    """Tests the `prepare.add_provider()` method.
+
+    Args:
+        json_data: JSON test file parsing fixture.
+        genome_file: Genome metadata JSON file.
+        gff3_file: GFF3 file.
+        output: Expected elements present in the updated genome metadata.
+        expectation: Context manager for the expected exception (if any).
+    """
+    genome_metadata = json_data(genome_file)
+    with expectation:
+        prepare.add_provider(genome_metadata, gff3_file)
+        for section, metadata in output.items():
+            for key, value in metadata.items():
+                assert genome_metadata[section].get(key, None) == value
+
+
+@pytest.mark.parametrize(
+    "genome_file, output",
+    [
+        ("genbank_genome.json", 2),
```

Comment:
Not entirely needed but  param set test ID here to conform to other tests. Also just tidies up the result table.

Reply:
My usual approach is to only add `id` when the print from pytest is going to look cluttered or too complicated to follow. In this case it felt pretty clean and clear (`test_prepare.py::test_add_assembly_version[genbank_genome.json-2]`), but happy to add an `id` if you feel it would be more readable :+1: